romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -52,16 +52,18 @@ class OutlierDetectionStep(RomanStep):
     def process(self, input_models):
         """Perform outlier detection processing on input data."""
 
-        self.input_models = input_models
-        self.input_container = isinstance(self.input_models, ModelContainer)
         self.skip = False
 
+        try:
+            self.input_models = ModelContainer(input_models)
+        except TypeError:
+            self.log.warning("Input cannot be parsed into a ModelContainer.")
+            self.skip = True
+            return
```

Comment:
This needs to return the input unchanged as the expectation is that the step returns the input if it cannot run so that the next step can still run in the pipeline.

Reply:
If `input_models` can't be parsed into a `ModelContainer`, that means it is invalid, in which case `finalize_results` will error out when trying to set metadata to invalid input when running an individual step:

https://github.com/spacetelescope/romancal/blob/879af02b536c354001df13198ce5a49af4637f52/romancal/stpipe/core.py#L66-L70

Reply:
I think there's contradiction in your comment because `None` is also not a parsable input for ModelContainer and currently the code returns `None`. 
All I am asking is to make the different return statements consistent. If you do so the test won't be looking for a generic Exception to be raised (which can be anything) but for catching the Warning that is issued when the TypeError is raised.

I will dismiss my review. Feel free to merge after considering the suggestions for the wording of the warnings.

Reply:
`OutlierDetectionStep` now returns the same input if it is not parsable into a `ModelContainer`. Thanks, @nden!